homeassistant/components/ultrasync/__init__.py

```diff
@@ -0,0 +1,136 @@
+"""The Interlogix/Hills ComNav UltraSync Hub component."""
+
+import asyncio
+
+from ultrasync import AlarmScene
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DATA_COORDINATOR,
+    DATA_UNDO_UPDATE_LISTENER,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    SENSORS,
+    SERVICE_AWAY,
+    SERVICE_DISARM,
+    SERVICE_STAY,
+)
+from .coordinator import UltraSyncDataUpdateCoordinator
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistantType, config: dict) -> bool:
+    """Set up the UltraSync integration."""
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up UltraSync from a config entry."""
+    if not entry.options:
+        options = {
+            CONF_SCAN_INTERVAL: entry.data.get(
+                CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+            ),
+        }
+        hass.config_entries.async_update_entry(entry, options=options)
+
+    coordinator = UltraSyncDataUpdateCoordinator(
+        hass,
+        config=entry.data,
+        options=entry.options,
+    )
+
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    undo_listener = entry.add_update_listener(_async_update_listener)
+
+    hass.data[DOMAIN][entry.entry_id] = {
+        DATA_COORDINATOR: coordinator,
+        DATA_UNDO_UPDATE_LISTENER: [undo_listener],
+        SENSORS: {},
+    }
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    _async_register_services(hass, coordinator)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+
+    if unload_ok:
+        for unsub in hass.data[DOMAIN][entry.entry_id][DATA_UNDO_UPDATE_LISTENER]:
+            unsub()
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+def _async_register_services(
```

Comment:
Why not create a `alarm_control_panel` entity? (https://developers.home-assistant.io/docs/core/entity/alarm-control-panel).

You can use the states `armed_away`, `armed_home` and `disarmed`. Creating your own implementation of this alarm is overkill (+ with alarm control panel you will benefit of nice extra features)

(An alarm_control_panel should be added as a follow up PR, since prs should contain just one platform).